source/History.js

```diff
@@ -0,0 +1,389 @@
+(function (enyo, scope) {
+	/**
+	* {@link moon.History} is a decorator that eables back key feature..
+	* When the object get back key input like pressing back key in R/C
+	* or history.back() with window.history object, it calls back key handler.
+	*
+	*
+	*/
+	moon.HistorySupport = {
+		/**
+		* @private
+		*/
+		name: 'HistorySupport',
+
+		published: {
+			/**
+			* When true, pressing back key makes panels close opened drawer
+			*
+			* @type {Bollean}
+			* @default true
+			* @public
+			*/
+			allowBackKey: true
+		},
+
+		/**
+		* @private
+		*/
+		events: {
+			onPushBackHistory: ''
+		},
+
+		/**
+		* @private
+		*/
+		handlers: {
+			/**
+			* Hanlder for back key input.
+			* To use custom behavior for back key, you can modify this handler.
+			*/
+			onPushBackHistory: 'pushBackHistory'
+		},
+
+		/**
+		* @private
```

Comment:
In your original implementation, developers could override this as needed, per control. Do you think it would be useful to maintain that feature? if so, I think we might want to mark this as `@public`.